app/services/initialize_session_service.rb

```diff
@@ -0,0 +1,55 @@
+#-- encoding: UTF-8
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+class InitializeSessionService
+  class << self
+    ##
+    # Initializes a new session for the given user.
+    # This services provides very little for what it is called,
+    # mainly caused due to the many ways a user can login.
+    def call(user, session)
+      session[:user_id] = user.id
+      session[:updated_at] = Time.now
+
+      if drop_old_sessions?
+        ::UserSession.where(user_id: user.id).delete_all
+      end
```

Comment:
In the services I created, I have sticked to a different interface for the services.

e.g. https://github.com/opf/openproject/blob/dev/app/services/update_work_package_service.rb

You would first have to instantiate a service, providing values for variables that are most likely unchanging even when calling the service a couple of times (e.g. work_package, user).

The `call` method would only take the changing part of the service. It would also return a service result:

https://github.com/opf/openproject/blob/dev/app/services/service_result.rb

While this might be too elaborated in this case, I'd still stick to that schema to have a unified interface.